src/OrleansTestKit/TestKitSilo.cs

```diff
@@ -41,6 +47,14 @@ public TestKitSilo()
         _grainRuntime = new TestGrainRuntime(GrainFactory, TimerRegistry, ReminderRegistry, ServiceProvider, StorageManager);
         ServiceProvider.AddService<IGrainRuntime>(_grainRuntime);
         _grainCreator = new TestGrainCreator(_grainRuntime, ServiceProvider);
+
+        var provider = new ServiceCollection()
+            .AddSingleton<GrainTypeResolver>()
+            .AddSingleton<IGrainTypeProvider, AttributeGrainTypeProvider>()
+            .AddSerializer()
+            .BuildServiceProvider();
+
+        _grainTypeResolver = provider.GetRequiredService<GrainTypeResolver>();
```

Comment:
Why do we need a DI container just to create this instance?

Reply:
Commented on the PR with a more full description.